### DISCUSSION TITLE
Question: BentoML configuration without yaml file

### DISCUSSION THREAD

**Discussion by chlee1016** [bentoml/BentoML#1756] [Q&A] [answered]
Hi team,

I am preparing to upgrade BentoML from version 0.11.0 to 0.13.1 and I have checked the breaking changes in configuration.
Previously, the docker image had run with a configuration using -e option.

**version 0.11.0**
```
$ docker run -e BENTOML__LOGGING__CONSOLE_LOGGING_ENABLED='true' -p 5000:5000 my-model
```

As I understand, setting configuration using -e option is deprecated in the latest version, so it should be replaced with yml file which is mounted to docker container.

**latest version (after 0.13.0)**
```
$ docker run -v /local/path/configuration.yml:/home/bentoml/configuration.yml -e BENTOML_CONFIG=/home/bentoml/configuration.yml -p 5000:5000 my-model
```

Q. Is there any way to directly set configuration using just one line command without writing configuration.yml file?

**Comment by parano** [comment, marked as answer]
Hi @chlee1016 - that is correct, we deprecated setting configs via env vars in favor of using the config file as the single source of truth for user-provided configurations. Is there a constraint in your deployment platform that makes it hard to provide configs via file?

**Reply by chlee1016** [reply]
Sorry for the late reply. We had to deploy the docker image by doing request to our internal docker server, so mounting config file needed more changes such as k8s configMap. Finally we decided to simply add the yml file into the bentoml service bundle. Thank you!

**Reply by parano** [reply]
cc @ssheng

**Reply by ssheng** [reply]
Thanks for sharing your use case, @chlee1016. We will incorporate it into our future design to make packaging configuration simpler.
